e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"kpt.dev/configsync/e2e"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	"kpt.dev/configsync/e2e/nomostest/taskgroup"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	"kpt.dev/configsync/pkg/kinds"
+	ocmetrics "kpt.dev/configsync/pkg/metrics"
+)
+
+const (
+	DefaultMonitorKSA = "default"
+	MonitorGSA        = "e2e-test-metric-writer"
+)
+
+func TestOtelCollectorDeployment(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.RequireGKE(t))
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml", "--ignore-not-found")
+		if t.Failed() {
+			nt.PodLogs("config-management-monitoring", ocmetrics.OtelCollectorName, "", false)
+		}
+	})
+
+	// If Workload Identity enabled on cluster, setup KSA to GSA annotation
+	if workloadPool, err := getWorkloadPool(nt); err != nil {
+		nt.T.Fatal(err)
+	} else if workloadPool != "" {
+		nt.T.Log("Workload identity enabled, adding KSA annotation to use %s service account", MonitorGSA)
+		nt.MustKubectl("annotate", "serviceaccount", "--namespace", ocmetrics.MonitoringNamespace, DefaultMonitorKSA,
+			fmt.Sprintf("iam.gke.io/gcp-service-account=%s@%s.iam.gserviceaccount.com", MonitorGSA, *e2e.GCPProject))
+	}
+
+	nt.T.Log("Restart otel-collector pod to refresh the ConfigMap, log and IAM")
+	nomostest.DeletePodByLabel(nt, "app", "opentelemetry", false)
+	if err := nt.WatchForAllSyncs(); err != nil {
```

Comment:
Killing the collector pod will drop the metrics database, right?
Or is it persisted now?
Do we need to wait for sync? Or can we just wait for the otel-collector to be Current?

Reply:
I suspect `WatchForAllSyncs` will just return immediately in this case, I don't think it's going to wait for the otel Pod to get restarted.

Reply:
Killing the pod should drop the metrics database. We are not validating any metric or value in this test, just checking if the exporting is clogged, so persistence is not hard requirement.

The repo should be already synced when the test started. We are just waiting for the new otel-collector deployment to be current, with the validateOtelCollectorStatusCurrent.

Reply:
In that case I think this `WatchForAllSyncs` can be removed

Reply:
Done!